public/js/main.js

```diff
@@ -645,6 +645,37 @@ function alignRightHeaders() {
     });
 }
 
+function addSingleQuoteToCsvIfNeeded(str) {
+  const specialChars = ['=', '+', '-', '@', '\t', '\r'];
+  if (specialChars.some(char => str.startsWith(char))) {
+    return "'" + str;
+  } else {
+    return str;
+  }
+}
+
+/**
+ * Patch the Tabulator CSV downloader logic to add a single quote to values starting with special characters
+ * This fixes the issue where Excel would interpret these values as formulas, creating a security risk
+ */
+function patchTabulatorCsvDownloader() {
+  const getFieldValueCopy = Tabulator.prototype.moduleBindings.download.prototype.getFieldValue;
+  Tabulator.prototype.moduleBindings.download.prototype.getFieldValue = function(columnName,rowData) {
+    const context = this;
+    let value = getFieldValueCopy.call(context, columnName, rowData);
+    switch (value === undefined ? "undefined" : typeof value) {
+      case 'object':
+        value = JSON.stringify(value);
+        break;
+      case 'undefined':
+      case 'null':
+        value = '';
+        break;
+    }
+    return addSingleQuoteToCsvIfNeeded(String(value));
```

Comment:
i’m thinking that you only need to call this function for string and number values (unless i misunderstood the code), why not check for this explicitly instead of trying to convert all other possible values to a string?
I’m unsure why we need to convert objects/null/undefined to strings ourselves btw.